src/log-pcap.c

```diff
@@ -1610,6 +1652,22 @@ static OutputInitResult PcapLogInitCtx(ConfNode *conf)
         }
     }
 
+    const char *bpf = conf == NULL ? NULL : ConfNodeLookupChildValue(conf, "bpf-filter");
+    if (bpf != NULL) {
+        /* Test compile the user provided BPF filter. */
```

Comment:
have you considered using `SCBPFCompile` here? Also, should we use the linktype of the capture device?

Reply:
Ah, `SCBPFCompile` would be better.

As for the link-type, we don't have it yet. This was meant to be a quick test to see if the bpf is valid at all, as we do compile it later when we do have the link type.  Perhaps we don't need to check it here, or if so just warn, and we can FatalError (or just warn) when opening the log files.

I guess that depends on if we consider a bad filter to be fatal, or just a warning and carry on as if there was no filter.

Reply:
Think we can be strict and just error out.